adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractVertxBasedCoapAdapter.java

```diff
@@ -745,15 +749,16 @@ protected final String getAuthId(final CoapExchange exchange) {
             return CompositeFuture.join(senderTracker, commandConsumerTracker)
             .compose(ok -> {
                 final DownstreamSender sender = senderTracker.result();
-                final Integer ttd = ttdTracker.result();
+                final Integer ttd = Optional.ofNullable(commandConsumerTracker.result())
+                        .map(c -> ttdTracker.result())
+                        .orElse(null);
                 final Message downstreamMessage = newMessage(
                         context.getRequestedQos(),
-                        ResourceIdentifier.from(endpoint.getCanonicalName(), context.getOriginDevice().getTenantId(),
-                                context.getOriginDevice().getDeviceId()),
+                        ResourceIdentifier.from(endpoint.getCanonicalName(), tenantId, deviceId),
                         "/" + context.getExchange().getRequestOptions().getUriPathString(),
                         contentType,
                         payload,
-                        tenantValidationTracker.result(),
+                        tenantTracker.result(),
```

Comment:
Is that change from `tenantValidationTracker` to `tenantTracker` intended?

Reply:
Yes. It makes the code more in line with the corresponding code from the HTTP adapter and makes it more obvious that it is about the returned `TenantObject` here. In the following lines the tenant object is accessed via `tenantTracker.result()` as well.

Reply:
Is there a test, which demonstrates, that a failure in `tenantValidationTracker` is handled proper by that code?

Reply:
A failure in the `tenantValidationTracker` would cause the `ttdTracker` to fail and subsequently the `commandConsumerTracker`. A failed `commandConsumerTracker` will cause the above code block (sending the telemetry/event message) to not be entered anyway.

A unit test in which a failed `tenantValidationTracker` is tested is `AbstractVertxBasedCoapAdapterTest#testMessageLimitExceededForAnEventMessage`.